climada/entity/exposures/crop_production.py

```diff
@@ -481,180 +490,201 @@ def init_full_exposure_set(input_dir=INPUT_DIR, filename=None, hist_mean_dir=HIS
 
-def normalize_with_fao_cropyield(exp_firr, exp_noirr, input_dir=INPUT_DIR,
-                                 yearrange=np.array([2008, 2018]),
-                                 unit='t', returns='all'):
-    """Normalize the given exposures countrywise with the mean cropyield production quantity
+def normalize_with_fao_cp(exp_firr, exp_noirr, input_dir=INPUT_DIR,
+                          yearrange=np.array([2008, 2018]), unit='t', return_data=True):
+    """Normalize the given exposures countrywise with the mean crop production quantity
     documented by the FAO.
 
-        Parameters:
-        exp_firr (cropyield_isimip): exposure under full irrigation
-        exp_noirr (cropyield_isimip): exposure under no irrigation
+    Parameters:
+        exp_firr (crop_production): exposure under full irrigation
+        exp_noirr (crop_production): exposure under no irrigation
+
+    Optional Parameters:
         input_dir (str): directory containing exposure input data
-        yearrange (array): the mean yield in this year range is used to normalize the exposure
-            data (default 2008-2018)
+        yearrange (array): the mean crop production in this year range is used to normalize
+            the exposure data (default 2008-2018)
         unit (str): unit in which to return exposure (t/y or USD/y)
-        returns (str): returned output
-            'all': country list, ratio = FAO/ISIMIP, normalized exposures, yield per country
-            as documented by the FAO and calculated by the ISIMIP dataset
-            else: country list, ratio = FAO/ISIMIP, normalized exposures
-
+        return_data (str): returned output
+            True: returns country list, ratio = FAO/ISIMIP, normalized exposures, crop production
+            per country as documented by the FAO and calculated by the ISIMIP dataset
+            False: country list, ratio = FAO/ISIMIP, normalized exposures
+
+    Returns:
+        country_list (list): List of country codes (numerical ISO3)
+        ratio (list): List of ratio of FAO crop production and aggregated exposure
+            for each country
+        exp_firr_norm (CropProduction): Normalized CropProduction (full irrigation)
+        exp_noirr_norm (CropProduction): Normalized CropProduction (no irrigation)
+
+    Returns (optional):
+        fao_crop_production (list): FAO crop production value per country
+        exp_tot_production(list): Exposure crop production value per country
+            (before normalization)
     """
 
-    # use the exposure in t/y to normalize with FAO yield values
-    if (exp_firr.value_unit == 'USD / y') and (exp_noirr.value_unit == 'USD / y'):
-        exp_firr.value = exp_firr.value_tonnes
-        exp_noirr.value = exp_noirr.value_tonnes
-    elif exp_firr.value_unit == 'USD / y':
+    # use the exposure in t/y to normalize with FAO crop production values
+    if exp_firr.value_unit == 'USD / y':
         exp_firr.value = exp_firr.value_tonnes
```

Comment:
I am a bit confused. Wasn't `value_tonnes` used above for `t / y` ?

Reply:
Yes. So the unit of 'value_tonnes' is t/y and this is the exposure the normalisation has to be applied to (not to an exposure with a value in USD/y). So for the case that the exposure is given in USD/y, it is "reset" to t/y if that makes sense?

Reply:
Precisely, maybe just make a small comment behind the line saying "temporary reset of value to t/y for normalization."

Reply:
It is very confusing why if the unit is actually in USD/y, you just replace the value by value_tonnes. Is there somewhere a value_usd ?

Reply:
To be more precise, I find it confusing that you have both an attribute `value_tonnes` and an attribute `value_unit`. Here is my reasoning: either, you have some `value` and `value_unit` which allow you to store both 't/y' or 'usd/y', or then use `value_tonnes` and discard `value_unit`.

Reply:
yes, i understand it is confusing. however, this is only a temporary thing happening within the normalization and never surfacing in the data. Carmen and me will figure out a way to make it less confusing for a reader of the code.

Reply:
So our idea was that value_unit contains the unit of the exposure value (either t/y or USD/y if the method set_to_USD() has been applied).  If the latter is the case, the initial value (in t/y) gets overwritten, but we store it in 'tonnes_per_year'. Temporarily resetting the value in this function allowed to treat all exposures the same, regardless of whether set_to_USD() had been applied beforehand. In the end this reset is reverted to return the exposure in the unit it was given. Do you have an idea on how to make this better understandable?

Reply:
I am sorry, but I really still do not understand what you are doing. So, the value of an exposure is defined in a certain unit, either t/y or USD/y. The unit value is stored in a variable `value_unit`. Hence, the variable for the value should not contain any reference to a value a be always in whatever value is given in `value_unit`. 

What does this sentence " If the latter is the case, the initial value (in t/y) gets overwritten, but we store it in 'tonnes_per_year'." mean? How can you convert the initial value into USD/y and then store it as t/y ?

If you have a function somewhere that requires all exposures to be in the same units, you could have for instance two variables, one `value_tonnes` and `value_in_unit`. The first is always set, and the second can be changed for whatever units you want. This has the disadvantage that the first might change without the second also being updated. But this could also be solved by making the second just a function of the first. However, this would required some computations everytime it is called...

Actually, the question is: what do you need the USD/y value for? Is it only for display?

Reply:
Regarding your final question: we require exposure in USD/y for risk assessments interested in the monetary value of the impacted crops rather than just tonnes. This conversion is offered as a method in the exposure because crop prices are country specific (otherwise we could just convert the impact with a single factor per crop type).

Regarding your proposal:
What you would call `value_in_unit` is actually precisely our `value`. For CLIMADA to work with this exposure, it always needs a column named 'value' and one attribute 'value_unit'.

So now here we have to options for value_unit: 't/y' and 'USD/y'.

If value_unit='t/y', the numbers in 'value' represent crop production per grid cell in 't/y'.

If value_unit='USD/y', the numbers in 'value' represent crop production per grid cell in 'USD/y' AND there is an extra column named 'value_tonnes' storing the crop production per grid cell in 't/y'. Reason: this is required for normalization.

Now, in the normalization method, IF value_unit='USD/y', the column value is temporarily re-set to the values of 'value_tonnes' and then reconverted to 'USD/y' at the end of the method call. This is temporary and only within the method and we do it because we normalize crop production with the help of national crop production statistics that come in t/y and not in USD/y...

I hope that helps to clarify?

Reply:
Yes, thanks a lot for your patience. Then maybe just rephrase the comment `# use the exposure in t/y to normalize with FAO crop production values` to make it clear that you temporarily reset the value and then put it back to USD/y. Maybe one could define a temp variable for this to be more clean, but not sure.

Reply:
So value_unit contains only the unit and value contains the actual exposure. 
Initially every exposure is in t/y. Hence value contains the exposure values and value_unit is 't/y'.
However the user might want to use the exposure in USD/y and can then use the method set_to_USD. This first saves the initial value in 'tonnes_per_year' (I changed the name as you proposed but haven't pushed it yet) and then it overwrites value with the value in USD/y and sets value_unit to 'USD/y'. (I had messed up the order earlier.)
Now for the normalisation we need the exposure in t/y because we normalise it with cropprodution values of the FAO in t/y. So in case the exposure is now in USD/y we temporarily set it back to the initial t/y exposure (which is still saved in tonnes_per_year). 
The reason why we set the actual value in set_to_USD is that we can then use all Climada functions (such as plotting functions or impact calculation that always refer to exp.value). 
Sorry for the confusion, I hope I explained it a bit better now.